addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1050,16 +1048,18 @@ def _set_last_seen_message(self, last_message, allow_older=False, notify=True):
         channel_member_domain = expression.AND([
             [('channel_id', 'in', self.ids)],
             [('partner_id', '=', current_partner.id) if current_partner else ('guest_id', '=', current_guest.id)],
-            [] if allow_older else expression.OR([
+            expression.OR([
+                [('new_message_separator_position', '<', last_message.id)],
                 [('seen_message_id', '=', False)],
-                [('seen_message_id', '<', last_message.id)]
+                [('seen_message_id', '<', last_message.id)],
             ])
         ])
         member = self.env['discuss.channel.member'].search(channel_member_domain)
         if not member:
             return
         member.write({
             'fetched_message_id': max(member.fetched_message_id.id, last_message.id),
+            'new_message_separator_position': 0,
```

Comment:
0 mean above all messages, shouldn't it be `last_message.id` + 1?

Reply:
I applied your suggestion to store the "first unseen" message id instead which simplify the code. So 0 means not shown at all.